File: src/main/scala/com/abaddon83/cardsGames/burracoGames/domainModels/Tris.scala
package com.abaddon83.cardsGames.burracoGames.domainModels

import com.abaddon83.cardsGames.shares.decks.Ranks.Rank
import com.abaddon83.cardsGames.shares.decks.{Card, Ranks}




trait Tris {
  protected val rank: Rank
  protected val cards: List[Card]

  def showRank(): Rank = rank
  def showCards(): List[Card] = cards

  def addCards(cardsToAdd: List[Card]): Tris

  def numCards():Int = {
    cards.size
  }

  protected def validateNewCards(cards: List[Card]): Tris

}

File: src/main/kotlin/com/abaddon83/burracoGame/writeModel/models/burracos/Tris.kt
package com.abaddon83.burracoGame.writeModel.models.burracos

import com.abaddon83.burracoGame.writeModel.models.decks.Card
import com.abaddon83.burracoGame.writeModel.models.decks.Ranks

abstract class Tris(className:String) :Burraco(className){
    protected abstract val rank: Ranks.Rank

    fun showRank(): Ranks.Rank = rank
    fun numCards():Int = cards.size


    abstract fun addCards(cardsToAdd: List<Card>): Tris
    protected abstract fun validateNewCards(cardsToValidate: List<Card>): List<Card>
}